chicken-sloths/bangazon-cli-sprint2
# DLK data generation

# Description
This PR pulls in our data faking modules created in [previous sprint](https://github.com/chicken-sloths/bangazon-api-sprint1/tree/master/data). Running `npm run data:generate` will generate `JSON` files of fake data (at `data/json`) for [each table](https://github.com/chicken-sloths/bangazon-cli-sprint2/wiki/Products-ERD). This PR includes fake data generated accordingly.

## Related Ticket(s)
N/A

## Problem to Solve
Provides fake data & modules for re-generating fake data, if necessary. Fake data provides something to build our tools around.

## Proposed Changes
Uses node package [faker](https://www.npmjs.com/package/faker)
Files are in `data/`
`data/build-data.js` is the main controller for creating the `JSON` files
`data/faker/index.js` orders the creation of `JSON` files since some fake data is dependent on others
`data/faker/_` are underscore because they are only ever called in `data/faker/index.js`

## Expected Behavior
`data/json` should include dope looking `JSON` files of data follow

## Steps to Test Solution
1. `npm install` (to install `faker`)
1. `npm run data:generate` should recreate/overwrite everything in `data/json`

## Documentation
- [x] _New documentation needs to be written_
